api/app/src/command/medical/patient/update-patient.ts

```diff
@@ -1,23 +1,68 @@
 import NotFoundError from "../../../errors/not-found";
-import { Patient } from "../../../models/medical/patient";
+import { PatientDataCommonwell } from "../../../external/commonwell/patient";
+import { Patient, PatientData } from "../../../models/medical/patient";
+import { getPatient } from "./get-patient";
 
-export const updatePatient = async ({
-  id,
+type PatientIdentifier = Pick<Patient, "id" | "cxId">;
+type PatientNoExternalData = Omit<PatientData, "externalData">;
+type PatientUpdate = PatientNoExternalData & PatientIdentifier;
+
+export const updatePatient = async (patient: PatientUpdate): Promise<Patient> => {
+  const { id, cxId } = patient;
+
+  // We don't want to update other fields, nor require the caller to send
+  // data that's not going to be updated, like `externalData`
+  const updatedPatient = await getPatient({ id, cxId });
+
+  const data = updatedPatient.data;
+  data.firstName = patient.firstName;
+  data.lastName = patient.lastName;
+  data.dob = patient.dob;
+  data.address = patient.address;
+  data.contact = patient.contact;
+
+  const [count] = await Patient.update(
+    {
+      data,
+    },
+    { where: { id, cxId } }
+  );
+  if (count < 1) throw new NotFoundError();
+  // TODO #156 Send this to Sentry
+  if (count > 1) console.error(`Updated ${count} patients for id ${id} and cxId ${cxId}`);
+
+  return updatedPatient;
+};
```

Comment:
This is returning the updated object, yes. We load it from the DB, update it with the data coming as parameter, send it to update the DB and return it.

We had `returning: true` so Sequelize would perform a select after updating the DB, but this adds another trip to the DB.

Since we're updating all data (not just some fields), I don't see a reason to query the DB after we update it. It if was a partial update, then I thing we should load after update.

Reply:
In this case it's just about the pattern that we set for updates in the code.

We're currently not using the field in the code, but for example `updated_at` will be changed, so the `Patient` we're returning isn't actually the updated patient reflected in the DB.

Kind of NDB for now, but worth keeping in mind since not returning the DB object could make the code more brittle (ie we add some update trigger down the road and have to remember to update this code)

Reply:
Added back